mysql-test/suite/s3/suite.pm

```diff
@@ -4,5 +4,45 @@ package My::Suite::S3;
 
 return "Need S3 engine" unless $::mysqld_variables{'s3'} eq "ON" or $ENV{HA_S3_SO};
 
+if (!$ENV{'S3_HOST_NAME'})
+{
```

Comment:
Before returning failure here, we've got our existing CI test suite using the MinIO so merging this code as is will fail our test suite.

So in the code where this github comment is, can you please:
* include client tcp connection to localhost:9000 (https://www.networkworld.com/article/2778731/connection-testing-with-perl.html)
* if that tcp works set the env vars to the values previously in mysql-test/suite/s3/my.cnf so the existing test runs don't need to change (additionally `S3_USE_HTTP=ON`))
* if tcp fails, return the text below.